src/WebJobs.Script.WebHost/Metrics/LinuxContainerLegionMetricsPublisher.cs

```diff
@@ -0,0 +1,251 @@
+        private readonly LegionMetricsFileManager _metricsFileManager;
+        private readonly TimeSpan _memorySnapshotInterval = TimeSpan.FromMilliseconds(1000);
+        private readonly TimeSpan _timerStartDelay = TimeSpan.FromSeconds(2);
+        private readonly IOptionsMonitor<StandbyOptions> _standbyOptions;
+        private readonly IDisposable _standbyOptionsOnChangeSubscription;
+        private readonly IEnvironment _environment;
+        private readonly ILogger _logger;
+        private readonly IMetricsLogger _metricsLogger;
+        private readonly string _containerName;
+
+        private TimeSpan _metricPublishInterval;
+        private Process _process;
+        private Timer _processMonitorTimer;
+        private Timer _metricsPublisherTimer;
+        private bool _initialized = false;
+
+        public LinuxContainerLegionMetricsPublisher(IEnvironment environment, IOptionsMonitor<StandbyOptions> standbyOptions, ILogger<LinuxContainerLegionMetricsPublisher> logger, IFileSystem fileSystem, ILinuxConsumptionMetricsTracker metricsTracker, IScriptHostManager scriptHostManager, int? metricsPublishIntervalMS = null)
+        {
+            _standbyOptions = standbyOptions ?? throw new ArgumentNullException(nameof(standbyOptions));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _metricsTracker = metricsTracker ?? throw new ArgumentNullException(nameof(metricsTracker));
+            if (scriptHostManager == null || !Utility.TryGetHostService<IMetricsLogger>(scriptHostManager, out _metricsLogger))
+            {
+                throw new InvalidOperationException("Unable to get IMetricsLogger Service.");
+            }
+            _containerName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.ContainerName);
+            _metricPublishInterval = TimeSpan.FromMilliseconds(metricsPublishIntervalMS ?? 30 * 1000);
+
+            // Default this to 15 minutes worth of files
+            int maxFileCount = 15 * (int)Math.Ceiling(1.0 * 60 / _metricPublishInterval.TotalSeconds);
+            string metricsFilePath = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsMetricsPublishPath);
+
+            _metricsFileManager = new LegionMetricsFileManager(metricsFilePath, fileSystem, logger, maxFileCount);
+
+            _processMonitorTimer = new Timer(OnProcessMonitorTimer, null, Timeout.Infinite, Timeout.Infinite);
+            _metricsPublisherTimer = new Timer(OnFunctionMetricsPublishTimer, null, Timeout.Infinite, Timeout.Infinite);
+
+            _metricsTracker.OnDiagnosticEvent += OnMetricsDiagnosticEvent;
+
+            if (_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                _standbyOptionsOnChangeSubscription = _standbyOptions.OnChange(o => OnStandbyOptionsChange());
```

Comment:
```suggestion
                _standbyOptionsOnChangeSubscription = _standbyOptions.OnChange(o => OnStandbyOptionsChange(o));
```

Might as well pass in options to the method as it is available.

Reply:
Done